addon/actions/action.js

```diff
@@ -32,6 +32,10 @@ export default EmberObject.extend({
   init() {
     this._super(...arguments);
     assert('Model has to be persisted!', !(this.get('instance') && !this.get('model.id')));
+
+    let payload = this.get('payload') || {};
+    assert('payload should be an object',  emberTypeOf(payload) === 'object');
```

Comment:
(restored comment)  
I've made an assert. Or maybe do you prefere another type of warning for user?
And maybe the same for queryParams?